core/src/main/java/com/google/cloud/sql/core/RefreshStrategy.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.core;
+
+/** Provide the refresh strategy to the DefaultConnectionInfoCache. */
+public interface RefreshStrategy {
+  ConnectionInfo getConnectionInfo(long timeoutMs);
+
+  void forceRefresh();
+
+  void refreshIfExpired();
```

Comment:
is `refreshIfExpired` needed on the base interface?

we don't have this on the Go interface i think https://github.com/GoogleCloudPlatform/cloud-sql-go-connector/pull/768?

Reply:
Yes. This is important for some edge cases in the RefreshAheadStrategy. It is a noop in the LazyRefresher.

Reply:
We do the equivalent in the dialer in the Go Connector. In theory, we could hide that inside the cache.